### ISSUE TITLE
Type.GetType() ignores assembly version in type name

### ISSUE THREAD

**Issue by bhaeussermann** [microsoft/vstest#4286]
## Description

When calling `System.Type.GetType()` from an NUnit test, the method will successfully return the named type, even if the version in the specified assembly qualified name does not match.

For example, 
```
Type.GetType("MyUtilities.Class1, MyUtilities, Version=1.1.0.0, Culture=neutral, PublicKeyToken=null", throwOnError: false)
```
will return the `Class1` type, even though the version of the `MyUtilities` assembly is 1.0.0, not 1.1.0

When I evaluate the same expression from a console application, the method will return `null` as expected.

This issue is preventing me from creating a test that fails when an assembly qualified name cannot be resolved due to a version mismatch.

## Steps to reproduce

- In a new solution, create a .NET Core Class Library named "MyUtilities"
- Add a new NUnit test project to the solution.
- Add a project reference to "MyUtilities" to the test solution.
- Add the following test:

```
namespace TestProject1;

[TestFixture]
internal class TestGetType
{
    [Test]
    public void Test()
    {
        Assert.IsNotNull(Type.GetType("MyUtilities.Class1, MyUtilities, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null", throwOnError: false));
        Assert.IsNull(Type.GetType("MyUtilities.Class1, MyUtilities, Version=1.1.0.0, Culture=neutral, PublicKeyToken=null", throwOnError: false));
    }
}
```

Run the test by executing the following command from the command line:

```
dotnet test TestProject1
```

## Expected behavior

The above test should pass.

## Actual behavior

The test fails since the second `Type.GetType()` call erroneously returns the type.

## Diagnostic logs

[log.txt](https://github.com/microsoft/vstest/files/10567949/log.txt)
[log.host.23-02-02_12-38-48_10956_5.txt](https://github.com/microsoft/vstest/files/10567952/log.host.23-02-02_12-38-48_10956_5.txt)

## Environment

OS: Windows 10 Enterprise, 10.0.19045 Build 19045
`dotnet --version`: 7.0.100
`dotnet test --version`: 17.4.0.51802
`Microsoft.NET.Test.Sdk` package version: 17.4.1

[AB#1744698](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1744698)

**Comment by MarcoRossignoli** [comment]
Can you try the same code with XUnit or MSTest? Do you have the same result?

Test platform and test adapters usually implements custom assembly resolver, so it's possible that the version is not checked in this chain of resolvers. 
If you confirm the same behavior for the other two brothers it's possible that the fallback to the test platform is wrongly loading, but we never had issue like that for what I know.

@nohwnd for more advice.

**Comment by bhaeussermann** [comment]
I tried the same code in an xUnit and an MSTest project and got the same result in both cases.

**Comment by nohwnd** [comment]
We do register AssemblyLoadContext.Resolve, but there is no way to opt out. Maybe creating a new AssemblyLoadContext. I am wondering what is the goal of this test? What are you trying to prove? The version of the assembly?

**Comment by bhaeussermann** [comment]
The test in question was added in response to a bug in a program. The program would read a fully qualified type name from a file and try to load the type using `Type.GetType()`. This was causing an exception because the loaded assembly had a different version from that specified in the fully qualified name.

In the spirit of TDD, I added a test to replicate the bug, expecting the test to fail. However, it didn't fail due to the issue at hand. Not only did this prevent me from following a strict test-driven process, but it is evident that if the program code which exhibited the bug (but is now fixed) were to regress at some point in future, the test will not necessarily fail to point out the bug.
